### PR TITLE
Fix table input defects

### PR DESCRIPTION
**Pull request by edleeks87**
# Description
Fixes issue with the width of date inputs when table is large, adds prop to isolate and toggle just the date width

### CODE REVIEW

File: src/components/table/table-cell/table-cell.spec.js [hunk 1]
```diff
@@ -122,11 +123,19 @@ describe('TableCell', () => {
         });
 
         it('renders a textarea that matches the expected style', () => {
-          wrapper = mount(
+          const wrapper1 = mount(
```
Comment (grabkowski) [inline src/components/table/table-cell/table-cell.spec.js]: why do we need another wrapper?

File: src/components/table/table-cell/table-cell.component.js [hunk 2]
```diff
@@ -30,17 +31,26 @@ class TableCell extends React.Component {
...
    * Returns if child element is Textarea.
    */
-  childrenHasTextArea() {
+  get checkChildrenInputType() {
     const { children } = this.props;
 
     if (!Array.isArray(children)) {
-      return this.childName(children) === TextArea.name;
+      if (this.childName(children) === TextArea.name) {
+        return 'isTextArea';
```
Comment (ianforster-sage) [inline src/components/table/table-cell/table-cell.component.js]: Please could you define these as constants rather than magic strings

Comment (edleeks87) [reply src/components/table/table-cell/table-cell.component.js]: Yeah I'll add them to the options helper

File: src/components/table/table-cell/table-cell.spec.js (hunk 1 above)
Comment (edleeks87) [reply src/components/table/table-cell/table-cell.spec.js]: I need to run two tests to meet coverage, I'll remove the extra wrapper and just reassign it the new component

**Pull request merged**